components/SearchBar.jsx

```diff
@@ -0,0 +1,80 @@
+/**
+ * SearchBar.jsx
+ * Created by Lizzie Salita 7/16/20
+ */
+
+import React, { useState, useEffect } from 'react';
+import { throttle } from 'lodash';
+import PropTypes from 'prop-types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { resetOrSubmitSearch, changeSearchTerm } from '../helpers/searchBarHelper';
+
+require('../styles/components/_searchBar.scss');
+
+const propTypes = {
+    setQuery: PropTypes.func.isRequired,
+    minChars: PropTypes.number
+};
+
+// the minimum number of characters a user is required to enter before they can perform a search
+const defaultProps = {
+    setQuery: () => {},
+    minChars: 2
+};
+
+const SearchBar = ({ setQuery, minChars }) => {
+    const [value, setValue] = useState('');
+    const [searchTerm, setSearchTerm] = useState('');
+
+    const resetSearch = () => {
+        setValue('');
+        setQuery('');
+        setSearchTerm('');
+    };
+
+    const onChange = throttle((e) => (changeSearchTerm(searchTerm, e) ? setValue(e.target.value) : resetSearch()), 500);
```

Comment:
This seems like a long time to throttle. Can we make this a param?

Reply:
Yeah, that is a good call.